File: AlgorithmicMDH-master/Matlab/Line.m
classdef Line
    %LINE 이 클래스의 요약 설명 위치
    %   자세한 설명 위치
    
    properties
        z
        z_hat
        p        
        p_perp
        v
        z_ceil
        v_ceil
    end
    
    methods
        function obj = Line(z,p)
            obj.z = reshape(z,[3,1]);
            obj.z_hat = reshape(normvec(z),[3,1]);
            obj.p = reshape(p,[3,1]);
            obj.v = cross(obj.p,obj.z_hat);
            obj.z_ceil=vecToso3(obj.z);
            obj.v_ceil=vecToso3(obj.v);
            obj.p_perp = cross(obj.z_hat,obj.v);
        end
        
    end
end
function ret = normvec(vec)
    ret = vec/norm(vec);
end

File: Python/Line.py
import math
import numpy as np
from numpy import cross,dot,sin,cos,arctan2
from numpy.linalg import norm

def vecToso3(x):
    ret = np.array([[0, -x[2], x[1]],
                    [x[2], 0, -x[0]],
                    [-x[1], x[0], 0]])
    return ret
def normvec(vec):
    ret = vec/norm(vec);
    return ret;
class Line:
    def __init__(self, z, p):
        self.z = np.reshape(np.array(z),[3]);
        self.z_hat = normvec(self.z);
        self.p = np.reshape(np.array(p),[3]);
        self.v = cross(self.p,self.z_hat);
        self.z_ceil = vecToso3(self.z_hat);
        self.v_ceil = vecToso3(self.v);
        self.p_perp= cross(self.z_hat , self.v);
    def print(self):
        print("direction vector : ",self.z)
        print("position vector : ",self.p)
        print("moment vector : ",self.v)
    def __str__(self):
        np.set_printoptions(suppress=True)
        str_="------------------------------------------------------\ndirection vector : "+str(np.array(self.z))+"\nposition vector : "+str(np.array(self.p))+"\nmoment vector : "+str(np.array(self.v))
        return str_